cli/formatter/table.go

```diff
@@ -0,0 +1,492 @@
+	for _, node := range batch {
+		if getNodeType(node) != t {
+			return false
+		}
+	}
+
+	return true
+}
+
+// renderScalars returns a table as string for scalars.
+func renderScalars(batch []any, transpose bool, opts Opts) (string, error) {
+	var arrays []any
+	for _, item := range batch {
+		arrays = append(arrays, []any{item})
+	}
+	return renderArrays(arrays, transpose, opts)
+}
+
+// isSingleArrayOfArrays returns true if the batch is a single array of arrays
+// and could be represented as a single table.
+func isSingleArrayOfArrays(batch []any) bool {
+	if len(batch) != 1 {
+		return false
+	}
+
+	if array, ok := batch[0].([]any); ok {
+		return isSingleType(array, arrayNodeType)
+	} else {
+		return false
+	}
+}
+
+// renderArrays returns a string representation of arrays.
+func renderArrays(batch []any, transpose bool, opts Opts) (string, error) {
+	if !isSingleArrayOfArrays(batch) {
+		return renderArraysAsTable(batch, transpose, opts)
+	} else {
+		return renderArraysAsTables(batch, transpose, opts)
+	}
+}
+
+// renderArraysAsTable returns a single table as string for the set of arrays.
+func renderArraysAsTable(batch []any, transpose bool, opts Opts) (string, error) {
+	maxLen := 0
+	for _, item := range batch {
+		itemLen := len(item.([]any))
+		if itemLen > maxLen {
+			maxLen = itemLen
+		}
+	}
+
+	var mapped []map[string]any
+	for _, item := range batch {
+		item := item.([]any)
+		itemMap := make(map[string]any)
+
+		for i := 0; i < maxLen; i++ {
+			if i < len(item) {
+				itemMap[strconv.Itoa(i+1)] = item[i]
+			} else {
+				itemMap[strconv.Itoa(i+1)] = ""
+			}
+		}
+
+		mapped = append(mapped, itemMap)
+	}
+
+	return renderEqualMaps(mapped, transpose, opts)
+}
+
+// renderArraysAsTables returns multiple tables as string for the set of the
+// set of arrays.
+func renderArraysAsTables(batch []any, transpose bool, opts Opts) (string, error) {
+	var result string
+	for _, node := range batch {
+		array := node.([]any)
+		barchRes, err := renderArraysAsTable(array, transpose, opts)
+		if err != nil {
+			return "", err
+		}
+		result = result + barchRes
+	}
+	return result, nil
+}
+
+// newTableWriter creates and configures new table writer.
+func newTableWriter(opts Opts) table.Writer {
+	t := table.NewWriter()
+	t.Style().Options.SeparateRows = true
+	if !opts.Graphics {
+		t.SetStyle(table.Style{Box: StyleWithoutGraphics})
+	}
+
+	return t
+}
+
+// handleColumnWidth handles width max value for tables columns.
+func handleColumnWidth(t table.Writer, columns int, opts Opts) {
+	colWidthTransformer := text.Transformer(func(val interface{}) string {
+		str := fmt.Sprintf("%v", val)
+		widthMax := opts.ColumnWidthMax
+		if utf8.RuneCountInString(str) > widthMax {
+			firstLine := string([]rune(str)[:widthMax])
+			remainingLines := string([]rune(str)[widthMax:])
+			firstLine = firstLine + "+"
+			remainingLines = text.InsertEveryN(remainingLines, '+', widthMax-1)
+			return firstLine + remainingLines
+		}
```

Comment:
Why do we need this, if we can just use `return text.InsertEveryN(str, '+', widthMax-1)`?

Reply:
It is a bit different output:
```
+-----+-----+-----+
| 7   | Eli | 18  |
|     | +za |     |
|     | +be |     |
|     | +th |     |
+-----+-----+-----+
```
vs
```
+-----+-----+-----+
| 7   | El+ | 18  |
|     | iz+ |     |
|     | ab+ |     |
|     | et+ |     |
|     | h   |     |
+-----+-----+-----+
```

Reply:
Is there any test for checking the correctness of handling the `ColumnWidthMax` argument? Maybe we should add one.

Reply:
Nevermind, here's this test https://github.com/tarantool/tt/blob/f3d4709c802e5bbdeaad691b8e1e215ad9e7b5aa/test/integration/connect/test_connect.py#L1471-L1502